blob/helper.go

```diff
@@ -37,11 +35,11 @@ func BlobsToShares(blobs ...*Blob) ([]share.Share, error) {
 	return shares.ToBytes(rawShares), nil
 }
 
-// ToAppBlobs converts node's blob type to the blob type from celestia-app.
-func ToAppBlobs(blobs ...*Blob) []*apptypes.Blob {
-	appBlobs := make([]*apptypes.Blob, 0, len(blobs))
+// ToAppBlobs converts node's blob type to the blob type from go-square.
+func ToAppBlobs(blobs ...*Blob) []*squareblob.Blob {
+	appBlobs := make([]*squareblob.Blob, 0, len(blobs))
 	for i := range blobs {
-		appBlobs[i] = &blobs[i].Blob
+		appBlobs[i] = blobs[i].Blob
```

Comment:
Off, that's a bug! We've slice with 0 length but non-zero capacity. Indexing will not work, this line should be `append` or `0` from `make` call should be dropped.

See https://go.dev/play/p/VBAeKkYcibD

Reply:
Resolved via cc3ee1dd91500e16ce25b4986a8cbb1d31e01b6d